### ISSUE TITLE
Limit on excluding domains from top lists

### ISSUE THREAD

**Issue by WaLLy3K** [pi-hole/AdminLTE#471]
**In raising this issue, I confirm the following:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*
- [x] The issue I'm reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**

4

---
**[BUG] Expected Behaviour:**
If putting in a large amount (50+) of domains within "Top Domains / Top Advertisers" (and possibly "Top Clients"), one expects to have domains removed from the Top Domains list on the dashboard.

**[BUG] Actual Behaviour:**
Domains after the 50th entry appear to be ignored, and are still listed on the Top Domains list on the dashboard

**[BUG] Steps to reproduce:**
Place 50 domains within the list found in Settings > API > Top Domains / Top Advertisers, and see whether domains after 50 are removed from the dashboard.

**Debug token generated by `pihole -d`:**
`h84i2r8laq`

**Comment by Mcat12** [comment]
Very odd, I can't reproduce on dev, so it might be fixed in the next update. [Here's](https://github.com/pi-hole/AdminLTE/blob/master/scripts/pi-hole/php/data.php#L144) the relevant code on master, which doesn't appear to be limiting the amount of domains you can remove.

**Comment by WaLLy3K** [comment]
This issue is present on the latest dev version on both my 3B running DietPi, and my Zero running RJL. With the Zero, I could reproduce the issue by copying my list over from the Admin Console and I could confirm the issue further by placing a domain from the bottom of the Top Domains API (that was showing on the dashboard Top Domains list) to the top.

**Comment by Mcat12** [comment]
Perhaps it's a formatting issue. Have you tried restarting FTL after making the change? Perhaps it's a bug in FTL, since that is what provides all the stats. I did see in your debug log that there were 51 excluded domains; is that how many you entered?

**Comment by WaLLy3K** [comment]
FTL's been restarted along with the box numerous times, and yeah, I've excluded 51 domains.

**Comment by DL6ER** [comment]
Now that I think about it again, I remember that I explicitly assumed that single lines in the `setupVars.conf` file are not longer than 1KB. Without reading anything I assume that you simply hit this limit. I'll increase the limit to something larger (maybe 100KB). Having said that I'm not in favor of a dynamical solution for the buffer size.

**Comment by WaLLy3K** [comment]
That sounds like it could be the case! I previously thought I hit a limit at around 47, but I removed a couple of domains and found I was able to add up to 50.

Assuming that 1KB is 50 domains, 100KB would eliminate any need to have a dynamic solution.

**Comment by DL6ER** [comment]
Try

```
awk '{++a[length()]} END{for (i in a) print i, a[i]}' /etc/pihole/setupVars.conf
```

To get the length of all lines in the config file and look for a line that is close to/longer than 1023

**Comment by WaLLy3K** [comment]
```
1075 1
11 1
12 1
20 1
13 1
21 2
22 2
23 1
16 1
24 1
18 1
26 1
28 1
37 1
0 2
```

**Comment by DL6ER** [comment]
There we are, one line with length 1075 has been found.

**Comment by DL6ER** [comment]
Fixed in https://github.com/pi-hole/FTL/commit/055ce53c23271ad9f8f56bc79ca0b6718684988e

Released as `FTL v2.3.1`
